### PR TITLE
Add timing and junit xml output to testsuite

### PR DESCRIPTION
**Pull request by sklam**
Changes testsuite to:
- Add custom logic to write junit xml output using `--junit`.
- Record test runtime and print them with `--show-timing`.
- Record PID and active tests per PID. Report these info at crash.

### CODE REVIEW

**Comment by gmarkall** [comment]
I tried clicking through to a test report (e.g. https://dev.azure.com/numba/numba/_TestManagement/Runs?runId=1926474&_a=runCharts) and I got a 401 (permission denied) error - are you able to access that? Do some permissions need changing somewhere?

**Comment by sklam** [comment]
> I tried clicking through to a test report (e.g. https://dev.azure.com/numba/numba/_TestManagement/Runs?runId=1926474&_a=runCharts) and I got a 401 (permission denied) error - are you able to access that? Do some permissions need changing somewhere?

I finally figured it out. I need to enable "Test Plans" feature in the project settings. The link now works!
You can also find the xml files in the attachments.

**Comment by gmarkall** [comment]
> > I tried clicking through to a test report (e.g. https://dev.azure.com/numba/numba/_TestManagement/Runs?runId=1926474&_a=runCharts) and I got a 401 (permission denied) error - are you able to access that? Do some permissions need changing somewhere?
> 
> I finally figured it out. I need to enable "Test Plans" feature in the project settings. The link now works! You can also find the xml files in the attachments.

I still get a 401 for that link :-(

**Comment by sklam** [comment]
Can you see the attachment the "Tests" tab:

Screenshot 2024-02-23 at 10 14 01 AM

Need to click on a test run (e.g. "JUnit_TestResults_NNNN") in the table then click "attachments" on the right panel

File: numba/tests/test_runtests.py [hunk 1]
```diff
@@ -279,6 +283,30 @@ def test_no_compilation_on_list(self):
...
+    def test_junit_output(self):
+        cmd = [sys.executable, '-m', 'numba.runtests', '--junit', '-vm1',
+               'numba.tests.test_usecases.TestUsecases.test_sum1d']
+        out = subprocess.check_output(cmd, encoding='utf8',
+                                      stderr=subprocess.STDOUT)
+        m = re.search(r"Writing junit xml file: (junit_numba_[0-9_-]+.xml)",
+                      out)
+        xml_filepath = m.group(1)
+        self.assertTrue(pathlib.Path(xml_filepath).exists())
+        # If this raise the XML file is invalid
```
Comment (esc) [inline numba/tests/test_runtests.py:307]:
```suggestion
        # If this raises, the XML file is invalid
```

**Review by esc — CHANGES_REQUESTED** [review]
Found a minor typo. Also, tried to log in to Azure to view the test results, as it appears as though I need to login to view them

This is what my screen looks like:

Screenshot 2024-02-28 at 23 08 02

**Comment by esc** [comment]
> Can you see the attachment the "Tests" tab:

Can you share more details about this screen and how to get there?

**Comment by esc** [comment]
OK, so there are some UI contortions that one has to go through, in order to retrieve the file, but it does seem to be possible w/o logging in.

Screenshot 2024-02-29 at 15 13 23

**Review by esc — APPROVED** [review]
Spellchecked and reviewed the code, looked at the generated RTD and worked out how to download and view the XML on azure.

LGTM

**Review by esc — APPROVED** [review]

**Pull request merged**